docs/Training-config.yml

```diff
@@ -0,0 +1,145 @@
+
+# define if preprocessing is performed?
+  preprocess: False
+
+# Define on what basis you want to split the input data folder for creating training image set, validation set and test set
+  split_by_column: pos_idx  # here 'pos_idx' indicates the data set was divided based on position list
+
+# Define the split ratio of data set
+  split_ratio:
+
+# here test:0.1 , train : 0.8, val:0.1, indicates 80% of images from the position list were used for training the model,
+# 10% were used for validation and 10% for testing
+    test: 0.1
+    train: 0.8
+    val: 0.1
+  squeeze: True
+  target_channels:
+  - 1
+  width: 256
+  random_seed: 0
+
+# there parameters defines the scales used for different augmentations used on input tiles for training the model
+  augmentations:
+    zoom_range: [0.6, 1.6]           # different magnification ranges the data can be in
+    rotate_range: 180                # 180 indicates all rotation degrees from 0 to 180 degrees
+    intensity_jitter: [0.5, 0.5]     # Jitter in image intensity to accomodate the varying intensities seen in images
+    noise_std: 0.5                   # Adds noise to images with standard deviation of 0.5
+    blur_range: [0, 10]              # Blurs the image from nil to range 10, accommodating conditions like images blurring at low magnification
+    shear_range: 10                  # Shear images to range of 10, accommodates cells stretching and deformation
+    
+
+# The upcoming parameters defines the microDL model used for training and required to set off training
+network:
+
+# Activation type shows activation layer type in the learning model, which helps process weighted inputs to deliver the output
+  activation:
+    type: relu    # relu is rectified linear unit, the activation function
+
+# specify if the input is batch normalized
+  batch_norm: True
+  block_sequence: conv-activation-bn
+
+# Specify if the model class is UNet2D or UNet2.5D or UNet3D. 3DUNet is heavy to run compared to 2.5D, thus making 2.5D more
+# efficient for use with 3D input data. 2DUNet is useful if using 2D data
+  class: UNet2D
+
+#
+  data_format: channels_first
+  dropout: 0.2
+  dropout_dense: 0.4
+  filter_size: 3
+  final_activation: linear
+  temperature: 1
+  height: 256
+  num_convs_per_block: 2
+
+# defines the number of filters used in each layer of the model, optimizing the weights as training progresses
+  num_filters_per_block:
+  - 16
+  - 32
+  - 64
+  - 128
+  - 256
+
+# Number of input channels, can be 1 if only phase images are input, can be 2 if both phase and retardance are used
+  num_input_channels: 1
+
+# Number of target channels defines the number of fluorescence channels to be predicted.
+  num_target_channels: 1     # 1 as only nuclear channels is to be predicted
+
+#
+  pooling_type: average
+  residual: True
+  skip_merge_type: concat
+  upsampling: bilinear
+  width: 256
+
+# Specify the parameters to define the sensitivity and speed with which to train the model.
```

Comment:
Set training parameters such as batch size, loss, and optimizer, as well as Keras callbacks.